### DISCUSSION TITLE
Proposal for better processors test cases in integration tests

### DISCUSSION THREAD

**Discussion by PhenX** [fluentmigrator/fluentmigrator#1940] [General]
While working on [this pull request](https://github.com/fluentmigrator/fluentmigrator/pull/1931) I found that the way different processors are tested in some of the integration tests could be improved by using TestCaseSource to treat each processor as a separate case : 

![image](https://github.com/user-attachments/assets/6742ff5d-0274-43de-982f-74eff31d6f18)

As you can see, each processor has its own result set, and I made it so that we still can specify unsupported processors on a test method (eg they won't appear as ignored test), or a list of specific processors to run on a test.

With all this, the test categories are also better supported, and auto generated in function of the test cases.

The changes are here : 

https://github.com/PhenX/fluentmigrator/commit/74c09187cf373e8e7f5194a4340c6b9102870cd3

Please note that I started from the branch in the Test containers pull request.

What do you think @jzabroski ?

**Comment by jzabroski** [comment]
I like it. It's a higher level of abstraction than xunit 's MemberDataAttribute in that the inputs describe which test they get injected to, rather than the test being responsible for knowing it's test parameters. Thinking about this, that is a core piece of why tools like CsCheck that do concolic testing are successful. But they have an even higher order Search function that searches an input space for optimal inputs.

Anyway, philosophical points aside, you can absolutely refactor this.

**Reply by PhenX** [reply]
Great ! thank you, I refactored all the uses of `ExecuteWithSupportedProcessors`, do you think we should/can go further with the tests in the `FluentMigrator.Tests.Integration.Processors` namespace ?

**Reply by PhenX** [reply]
Also, I can't make a pull request with this while the other one with test container is not merged though.

**Reply by jzabroski** [reply]
the test container one was marked as draft?

**Reply by PhenX** [reply]
Oh I forgot this 😅
Not draft anymore

**Reply by jzabroski** [reply]
oh ok. I may have time to review today. planning to ship 6.3 today
